pkgs/by-name/ed/edopro/package.nix

```
@@ -0,0 +1,299 @@
+      runHook preInstall
+
+      mkdir -p $out/{bin,include}
+      cp lib/Linux/libIrrlicht.a $out/bin
+      cp -r include/* $out/include
+
+      runHook postInstall
+    '';
+  };
+
+  ocgcore = stdenv.mkDerivation rec {
+    pname = "ygopro-core";
+    version = deps.ocgcore-rev;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "ygopro-core";
+      rev = version;
+      hash = deps.ocgcore-hash;
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [ lua5_3 ];
+
+    preBuild = ''
+      premake5 gmake2
+    '';
+
+    enableParallelBuilding = true;
+    buildFlags = [
+      "verbose=true"
+      "config=release"
+      "ocgcore"
+    ];
+    makeFlags = [
+      "-C"
+      "build"
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      cp bin/release/libocgcore.a $out/bin
+
+      runHook postInstall
+    '';
+  };
+
+  edopro = stdenv.mkDerivation rec {
+    pname = "edopro";
+    version = deps.edopro-version;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "edopro";
+      rev = deps.edopro-rev;
+      hash = deps.edopro-hash;
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      bzip2
+      curl
+      flac
+      fmt_9
+      freetype
+      irrlicht-edopro
+      libevent
+      libgit2
+      libjpeg
+      libpng
+      libvorbis
+      nlohmann_json
+      openal
+      SDL2
+      sqlite
+    ];
+
+    # nixpkgs' gcc stack currently appears to not support LTO
+    postPatch = ''
+      substituteInPlace premake5.lua \
+        --replace-fail 'flags "LinkTimeOptimization"' 'removeflags "LinkTimeOptimization"'
+    '';
+
+    preBuild = ''
+      premake5 gmake2 \
+        --covers=\"${covers_url}\" \
+        --fields=\"${fields_url}\" \
+        --no-core \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${ocgcore}/bin" \
+        --sound=sfml
+    '';
+
+    enableParallelBuilding = true;
+    env = {
+      CXXFLAGS = "-include cstdint";
+      LDFLAGS = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    };
+    buildFlags = [
+      "verbose=true"
+      "config=release_x64"
+      "ygoprodll"
+    ];
+    makeFlags = [
+      "-C"
+      "build"
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      cp bin/x64/release/ygoprodll $out/bin
+
+      runHook postInstall
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        # Needed if we download files from ProjectIgnis' website or any https only website.
+        "cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "EDOPro" ''
```

So it fits to `meta.mainProgram`.

```suggestion
    writeShellScriptBin "edopro" ''
```